### PR TITLE
write catalog now writes relative paths per default

### PR DESCRIPTION
**Pull request by francescalb**
It is now possible to choose to write abslute paths when writing catalog, but relative paths to current folder are default.

# Description:

## Type of change:

- [ ] Bug fix.
- [x] New feature.
- [ ] Documentation update.

## Checklist:

This checklist can be used as a help for the reviewer.

- [ ] Is the code easy to read and understand?
- [ ] Are comments for humans to read, not computers to disregard?
- [ ] Does a new feature has an accompanying new test (in the CI or unit testing schemes)?
- [ ] Has the documentation been updated as necessary?
- [ ] Does this close the issue?
- [ ] Is the change limited to the issue?
- [ ] Are errors handled for all outcomes?
- [ ] Does the new feature provide new restrictions on dependencies, and if so is this documented?

### CODE REVIEW

File: ontopy/utils.py [hunk 1]
```diff
@@ -391,12 +391,22 @@ def write_catalog(
             (URIs).  It has the same format as the dict returned by
             read_catalog().
         output: name of catalog file.
-        dir: directory path to the catalog file.  Only used if `output`
+        directory: directory path to the catalog file.  Only used if `output`
             is a relative path.
+        relative_paths: whether to write aboslute paths or paths relative to
```
Comment (jesper-friis) [inline ontopy/utils.py:396]: whether to write relative or absolute paths to...

File: ontopy/utils.py [hunk 2]
```diff
@@ -391,12 +391,22 @@ def write_catalog(
...
             If false, an existing file will be overwritten.
     """
-    filename = (Path(dir) / output).resolve()
+    web_protocol = "http://", "https://", "ftp://"
+    if relative_paths:
+        for key, item in mappings.items():
+            mappings[key] = (
+                item
+                if item.startswith(web_protocol)
+                else os.path.relpath(item, Path(directory).resolve())
```
Comment (jesper-friis) [inline ontopy/utils.py:407]:
```suggestion
            if not item.startswith(web_protocol):
                mappings[key] = os.path.relpath(item, Path(directory).resolve())
```

File: tests/test_catalog.py [hunk 3]
```diff
@@ -61,6 +64,66 @@ def test_catalog(repo_dir: "Path", tmpdir: "Path") -> None:
...
+    write_catalog(catalog, output=tmp_catalog_path, relative_paths=False)
     tmp_catalog = read_catalog(str(tmp_catalog_path))
     assert tmp_catalog == catalog
+
+
+def test_write_catalog_choosing_relative_paths(
+    repo_dir: "Path", tmpdir: "Path"
+) -> None:
+    ontodir = repo_dir / "tests" / "catalogs_for_testing"
+    print(" PRINT ")
```
Comment (jesper-friis) [inline tests/test_catalog.py:76]:
```suggestion
```

**Review by jesper-friis — COMMENTED** [review]
Looks good, but haven't tested it... Take a look at the comments and apply those that make sense.

File: ontopy/utils.py (hunk 2 above)
Comment (francescalb) [reply ontopy/utils.py:407]: Added by means of script, for prior testing.

**Review by jesper-friis — APPROVED** [review]

**Pull request merged**